### ISSUE TITLE
pinned message in callback function

### ISSUE THREAD

**Issue by TurtleOld** [eternnoir/pyTelegramBotAPI#2027]
1. What version of pyTelegramBotAPI are you using? **4.12.0**

2. What OS are you using? **Ubuntu 22.04**

3. What version of python are you using? 3.10.12

Hey there everyone!

I'm trying to pin the bot's message in a personal correspondence with the user in the callback function. The message is pinned, but an error appears:
```
Traceback (most recent call last):
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 955, in infinity_polling
    self.polling(non_stop=True, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 1046, in polling
    self.__non_threaded_polling(non_stop=non_stop, interval=interval, timeout=timeout, long_polling_timeout=long_polling_timeout,
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 1179, in __non_threaded_polling
    raise e
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 1144, in __non_threaded_polling
    self.__retrieve_updates(timeout, long_polling_timeout, allowed_updates=allowed_updates)
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 652, in __retrieve_updates
    self.process_new_updates(updates)
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 740, in process_new_updates
    self.process_new_messages(new_messages)
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 775, in process_new_messages
    self._notify_command_handlers(self.message_handlers, new_messages, 'message')
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 6863, in _notify_command_handlers
    self._exec_task(
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 1198, in _exec_task
    raise e
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 1191, in _exec_task
    task(*args, **kwargs)
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 6766, in _run_middlewares_and_handler
    if self._test_message_handler(handler, message):
  File "/home/alexander/.cache/pypoetry/virtualenvs/RoV0BKea-py3.10/lib/python3.10/site-packages/telebot/__init__.py", line 6676, in _test_message_handler
    for message_filter, filter_value in message_handler['filters'].items():
TypeError: 'function' object is not subscriptable

``` 

```
@bot_admin.callback_query_handler(
    func=lambda call: call.data.startswith(
        'select_account_',
    ),
)
def handle_select_account(call):
    account_id = int(call.data.split('_')[2])
    account = Account.objects.filter(id=account_id).first()
    if account:
        telegram_user_id = call.from_user.id
        telegram_user = TelegramUser.objects.filter(
            telegram_id=telegram_user_id,
        ).first()
        if telegram_user:
            telegram_user.selected_account_id = account_id
            telegram_user.save()
            bot_admin.unpin_all_chat_messages(chat_id=call.message.chat.id)
            pinned_message = bot_admin.send_message(
                chat_id=call.message.chat.id,
                text=f'Selected account: {account.name_account}',
            )
            bot_admin.pin_chat_message(
                chat_id=call.message.chat.id,
                message_id=pinned_message.message_id,
                disable_notification=True,
            )
        else:
            bot_admin.send_message(
                call.message.chat.id,
                'Error: account not selected.',
            )
```
It looks very much like the problem lies in callback_query_handler. But why and how to fix it, I can't find a solution in any way. I will be grateful for any help in understanding this error and how to solve it
I tried to install version 4.6.0, the same error.

**Comment by TurtleOld** [comment]
```
pinned_message = bot_admin.send_message(
    chat_id=call.message.chat.id,
    text=f'Selected account: {account.name_account}',
)
bot_admin.pin_chat_message(
    chat_id=call.message.chat.id,
    **message_id=pinned_message.id,**
    disable_notification=True,
)
```

**Comment by TurtleOld** [comment]
```
to_pin = bot_admin.send_message(
    chat_id=call.message.chat.id,
    text=f"Выбран счёт: {account.name_account}",
)
print("to_pin:", to_pin.message_id)
bot_admin.pin_chat_message(
    chat_id=call.message.chat.id,
    message_id=to_pin.message_id,
    disable_notification=True,
)
```

I found out that the problem lies in the `pin_chat_message method`. If we make the command `print("to_pin:", to_pin.message_id)`, we will get to_pin: 7929

**Comment by Badiboy** [comment]
Try
```
@bot_admin.callback_query_handler(
    func=lambda call: call.data and call.data.startswith(
        'select_account_',
    ),
)
```
?

**Comment by TurtleOld** [comment]
@Badiboy. unfortunately, it didn't help.
I came to the conclusion that there is still a problem with `callback_query_handler,` because I sent a request through the browser and the same `TypeError` returned to me
https://api.telegram.org/bot<token>/pinChatMessage?chat_id=<chat_id>&message_id=<message_id>

But eventually I came to the conclusion that there is no problem at all. I am trying to pin a message in the user's correspondence with the bot. And pin_chat_message works with a supergroup. I tested it in a group, there is no error there. I think if there are no other solutions, close the issue.
